roles/confluent.variables_handlers/defaults/main.yml

```
@@ -262,12 +262,16 @@ ksql:
 ksql_jolokia_enabled: "{{jolokia_enabled}}"
 ksql_jolokia_port: 7774
 ksql_jolokia_ssl_enabled: "{{ ksql_ssl_enabled }}"
+ksql_processing_log: "processing_log"
+ksql_log_streaming_enabled: false
```

should this be turned on by default? i think thats the ksql recommendation

---

I spoke with Product about this, and due to the way that Log Streaming works with RBAC, Product made the decision to disable it by default as it adds far more overhead to the end user to setup.  The idea being, giving the user the choice to go through that overhead if they really want this feature.  It's directly related to my previous comment about the listener and the document I shared on slack.

---

Ah yes that makes sense about rbac and the listener because the interbroker listener can't be oauth. 

Seems to me a simpler solution would be just to not have ksql_log_streaming_enabled at all if rbac is enabled:
```suggestion
ksql_log_streaming_enabled: "{{ not rbac_enabled }}"
```

---

@domenicbove I've spoken with the KSQL team and this feature is expected to function in this way with RBAC.  As in, this feature should be available even when RBAC is enabled and use another listener.

---

Dang, really wish we could just say "no ksql log streaming if you have rbac on", because it just feels so unnatural to connect to the interbroker listener instead of the ksql listener. 

Wasn't there a whole security patch that said if rbac is on each component must connect over oauth?
I remember the operator team had to struggle w that and we were better off bc we didn't allow that.

---

Completely agree with you!  It maybe worth discussing this again with Product to get a final say on how to proceed here.  A few other engineers have also been surprised with the configuration around this as well.